TBC1D3.py

```diff
@@ -0,0 +1,299 @@
+import os
+from typing import Union
+from Bio import SeqIO, SeqUtils
+from abc import ABC, abstractmethod
+
+
+class NucleotideNotFoundError(ValueError):
+    """
+    Error raised when there is incorrect nucleotide in corresponding sequence.
+    """
+
+    pass
+
+
+class AminoAcidNotFoundError(ValueError):
+    """
+    Error raised when there is incorrect aminoacid in corresponding sequence.
+    """
+
+    pass
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__(self):
+        """Return the length of the biological sequence."""
+        pass
+
+    def __getitem__(self, item):
+        """Return the element or slice of the biological sequence."""
+        pass
+
+    def __str__(self):
+        """Return the string representation of the biological sequence."""
+        pass
+
+    def is_correct(self):
+        """Check if the alphabet of the sequence is correct."""
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence, ABC):
```

Comment:
Здесь можно уже не наследоваться от ABC т.к. ABC уже есть в BiologicalSequence :)
```suggestion
class NucleicAcidSequence(BiologicalSequence):
```